src/libraries/System.Text.Json/src/PACKAGE.md

```diff
@@ -1,11 +1,258 @@
 
-The `System.Text.Json` library is built-in as part of the shared framework in .NET Runtime. The package can be installed when you need to use it in other target frameworks.
+## How to Use
+
+<!-- A compelling example on how to use this package with code, as well as any specific guidelines for when to use the package -->
+The System.Text.Json library is built-in as part of the shared framework in .NET Runtime. The package can be installed when you need to use it in other target frameworks.
+
+Serialization:
+```csharp
+using System;
+using System.Text.Json;
+
+WeatherForecast forecast = new (DateTimeOffset.Now, 26.6f, "Sunny");
+var serialized = JsonSerializer.Serialize(forecast);
+
+Console.WriteLine(serialized);
+// {"Date":"2023-08-02T16:01:20.9025406+00:00","TemperatureCelsius":26.6,"Summary":"Sunny"}
+
+var forecastDeserialized = JsonSerializer.Deserialize<WeatherForecast>(serialized);
+Console.WriteLine(forecast == forecastDeserialized);
+// True
+
+public record WeatherForecast(DateTimeOffset Date, float TemperatureCelsius, string? Summary);
+```
+
+Serialization using the source generator:
+```csharp
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+WeatherForecast forecast = new (DateTimeOffset.Now, 26.6f, "Sunny");
+var serialized = JsonSerializer.Serialize(forecast, SourceGenerationContext.Default.WeatherForecast);
+
+Console.WriteLine(serialized);
+// {"Date":"2023-08-02T16:01:20.9025406+00:00","TemperatureCelsius":26.6,"Summary":"Sunny"}
+
+var forecastDeserialized = JsonSerializer.Deserialize<WeatherForecast>(serialized, SourceGenerationContext.Default.WeatherForecast);
+Console.WriteLine(forecast == forecastDeserialized);
+// True
+
+public record WeatherForecast(DateTimeOffset Date, float TemperatureCelsius, string? Summary);
+
+[JsonSourceGenerationOptions(WriteIndented = true)]
+[JsonSerializable(typeof(WeatherForecast))]
+internal partial class SourceGenerationContext : JsonSerializerContext
+{
+}
+```
+
+Using the JSON DOM:
+```csharp
+
+using System;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+
+string jsonString =
+@"{
+  ""Date"": ""2019-08-01T00:00:00"",
+  ""Temperature"": 25,
+  ""Summary"": ""Hot"",
+  ""DatesAvailable"": [
+    ""2019-08-01T00:00:00"",
+    ""2019-08-02T00:00:00""
+  ],
+  ""TemperatureRanges"": {
+      ""Cold"": {
+          ""High"": 20,
+          ""Low"": -10
+      },
+      ""Hot"": {
+          ""High"": 60,
+          ""Low"": 20
+      }
+  }
+}
+";
+
+JsonNode forecastNode = JsonNode.Parse(jsonString)!;
+
+
+// Get value from a JsonNode.
+JsonNode temperatureNode = forecastNode!["Temperature"]!;
```

Comment:
```suggestion
JsonNode temperatureNode = forecastNode["Temperature"]!;
```

Reply:
All these came from the official conceptual docs - so we may want to follow up to get those fixed: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/use-dom#use-jsonnode